File: ios/ShortDrama/Sources/Domain/UseCases/SendOtpUseCase.swift
import Foundation

struct SendOtpUseCase: Sendable {
    private let repository: AuthRepositoryProtocol

    init(repository: AuthRepositoryProtocol) {
        self.repository = repository
    }

    func execute(phone: String, countryCode: String, scene: String) async throws -> SendOtpResult {
        try await repository.sendOtp(phone: phone, countryCode: countryCode, scene: scene)
    }
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/usecase/SendOtpUseCase.kt
package com.djs66256.short_drama.domain.usecase

import com.djs66256.short_drama.core.network.ApiResult
import com.djs66256.short_drama.data.dto.SendOtpResult
import com.djs66256.short_drama.domain.repository.AuthRepository
import javax.inject.Inject

class SendOtpUseCase @Inject constructor(
    private val authRepository: AuthRepository,
) {
    suspend operator fun invoke(
        countryCode: String,
        phone: String,
        scene: String,
    ): ApiResult<SendOtpResult> {
        return authRepository.sendOtp(
            countryCode = countryCode,
            phone = phone,
            scene = scene,
        )
    }
}
